manifest.md

```diff
@@ -0,0 +1,261 @@
+<!--[metadata]>
++++
+draft = true
++++
+<![end-metadata]-->
+
+# Image Manifest
+
+The first goal is content-addressable images, by supporting an image model where the image's configuration can be hashed to generate an ID for the image and its components.
+The second goal is to allow multi-architecture images, through a "fat manifest" which references image manifests for platform-specific versions of an image.
+The third goal is to be translatable to the [OpenContainers/runtime-spec](https://github.com/opencontainers/runtime-spec)
+
+# Media Types
+
+The following media types are used by the manifest formats described here, and the resources they reference:
+
+- `application/vnd.oci.image.manifest.v1+json`: Image manifest format
+- `application/vnd.oci.image.manifest.list.v1+json`: Manifest list, aka "fat manifest"
+- `application/vnd.oci.image.serialization.v1+json`: Container config JSON
+- `application/vnd.oci.image.rootfs.tar.gzip`: "Layer", as a gzipped tar
+
+## Manifest List
+
+The manifest list is the "fat manifest" which points to specific image manifests for one or more platforms.
+Its use is optional, and relatively few images will use one of these manifests.
+A client will distinguish a manifest list from an image manifest based on the Content-Type returned in the HTTP response.
+
+## *Manifest List* Field Descriptions
+
+- **`schemaVersion`** *int*
+
+  This field specifies the image manifest schema version as an integer. This
+  schema uses the version `2`.
+
+- **`mediaType`** *string*
+
+  The MIME type of the manifest list. This should be set to
+  `application/vnd.oci.image.manifest.list.v1+json`.
+
+- **`manifests`** *array*
+
+  The manifests field contains a list of manifests for specific platforms.
+
+  Fields of a object in the manifests list are:
+
+  - **`mediaType`** *string*
+
+    The MIME type of the referenced object. This will generally be
+    `application/vnd.oci.image.manifest.v1+json`, but it could also
+    be `application/vnd.oci.image.manifest.v1+json` if the manifest
+    list references a legacy schema-1 manifest.
+
+  - **`size`** *int*
```

Comment:
@duglin `Content-Length` is typically insufficient for this purpose. In addition, this specification cannot and should not assume that the underlying distribution mechanism knows about the complete size.

It is much more efficient to mark this during assembly. This is mostly just the size of the manifest json, which is usually small and at hand anyways.